observability-test/spanner.ts

```diff
@@ -0,0 +1,367 @@
+    let spannerMock: mock.MockSpanner;
+    let traceExporter: typeof InMemorySpanExporter;
+
+    const contextManager = new AsyncHooksContextManager();
+    setGlobalContextManager(contextManager);
+
+    afterEach(() => {
+      disableContextAndManager(contextManager);
+    });
+
+    beforeEach(async () => {
+      const setupResult = await setup();
+      spanner = setupResult.spanner;
+      server = setupResult.server;
+      spannerMock = setupResult.spannerMock;
+
+      const selectSql = 'SELECT 1';
+      const updateSql = 'UPDATE FOO SET BAR=1 WHERE BAZ=2';
+      spannerMock.putStatementResult(
+        selectSql,
+        mock.StatementResult.resultSet(createSelect1ResultSet())
+      );
+      spannerMock.putStatementResult(
+        updateSql,
+        mock.StatementResult.updateCount(1)
+      );
+
+      traceExporter = new InMemorySpanExporter();
+      const sampler = new AlwaysOnSampler();
+
+      const provider = new NodeTracerProvider({
+        sampler: sampler,
+        exporter: traceExporter,
+      });
+      provider.addSpanProcessor(new SimpleSpanProcessor(traceExporter));
+
+      const instance = spanner.instance('instance');
+      database = instance.database('database');
+      database.observabilityConfig = {
+        tracerProvider: provider,
+        enableExtendedTracing: false,
+      };
+    });
+
+    afterEach(() => {
+      traceExporter.reset();
+      spannerMock.resetRequests();
+      spanner.close();
+      server.tryShutdown(() => {});
+    });
+
+    it('getSessions', async () => {
+      const [rows] = await database.getSessions();
+
+      traceExporter.forceFlush();
+      const spans = traceExporter.getFinishedSpans();
+      assert.strictEqual(spans.length, 1, 'Exactly 1 span expected');
+
+      const actualSpanNames: string[] = [];
+      spans.forEach(span => {
+        actualSpanNames.push(span.name);
+      });
+
+      const expectedSpanNames = ['CloudSpanner.Database.getSessions'];
+      assert.deepStrictEqual(
+        actualSpanNames,
+        expectedSpanNames,
+        `span names mismatch:\n\tGot:  ${actualSpanNames}\n\tWant: ${expectedSpanNames}`
+      );
+    });
+
+    it('getSnapshot', done => {
+      database.getSnapshot((err, transaction) => {
+        assert.ifError(err);
+
+        transaction!.run('SELECT 1', (err, rows) => {
+          assert.ifError(err);
+
+          traceExporter.forceFlush();
+          const spans = traceExporter.getFinishedSpans();
+          assert.strictEqual(spans.length, 1, 'Exactly 1 span expected');
+
+          const actualSpanNames: string[] = [];
+          spans.forEach(span => {
+            actualSpanNames.push(span.name);
+          });
+
+          const expectedSpanNames = ['CloudSpanner.Database.getSnapshot'];
+          assert.deepStrictEqual(
+            actualSpanNames,
+            expectedSpanNames,
+            `span names mismatch:\n\tGot:  ${actualSpanNames}\n\tWant: ${expectedSpanNames}`
+          );
+
```

Comment:
Add checks for events as well in E2E tests

Reply:
Done, thank you @surbhigarg92!